### DISCUSSION TITLE
Support for ALPR (CodeProject.AI)

### DISCUSSION THREAD

**Discussion by katoog** [VorlonCD/bi-aidetection#332] [Ideas]
It would be great to be able to get plate recognition and have the plate number shown in Telegram at the same time as the object.
e.g. "Driveway - Car (75%) - Plate: ABC123"

**Comment by VorlonCD** [comment]
Try this guy:  https://github.com/VorlonCD/bi-aidetection/blob/master/src/UI/Installer/AIToolSetup.2.5.22.exe

Supports all Codeproject.ai server options including ALPR.

I did notice when I compared CPAI ALPR with Sighthound.vehicle, that sighthound was much better at detecting plate, etc on some images.

![image](https://github.com/VorlonCD/bi-aidetection/assets/7512113/781fbe92-ee0d-40a3-8d69-ab2909896245)

**Comment by katoog** [comment]
You sir, are quick! Thank you.

I have installed it and added the CodeProject AI (License Plate) server. I did a quick test on a clear license plate and it didn't get it exactly right (missing the first and last letter), but I appreciate this is a CodeProject issue and will continue to play with it.

I'm getting spammed this in Telegram when it can't detect a plate. How do I suppress it?

`ERROR> Failure response from 'CodeProject_AI_Plate'. Error='No plates found'. JSON: '{"success":true,"predictions":[],"message":"No plates found","processMs":299,"inferenceMs":290,"code":200,"analysisRoundTripMs":351}'`

And a second messsage:
`ERROR> ... AI URL failed with ' Failure response from 'CodeProject_AI_Plate'. Error='No plates found'. JSON: '{"success":true,"predictions":[],"message":"No plates found","processMs":299,"inferenceMs":290,"code":200,"analysisRoundTripMs":351}'' - for 'CodeProject_AI_Plate' failed '11' times.  Disabling: 'http://127.0.0.1:32168/v1/image/alpr'
`

Again, thank you for adding CodeProject AI support :)  If you accept Bitcoin I'd be happy to donate.

**Comment by VorlonCD** [comment]
This fixes the error:
https://github.com/VorlonCD/bi-aidetection/blob/master/src/UI/Installer/AIToolSetup.2.5.24.exe

Also, Sighthound is free.  It allows 5000 queries a month and provides the vehicle color, model, plate number and STATE for the plate if it can see it well enough.   It also has better plate detection - I had several images that CodeProject did not detect but Sighthound did.  Sign up here and click NEW TOKEN.   Then give AITOOL the token when prompted.

https://accounts.sighthound.com/#/sighthound-cloud

I haven't gone over 5000 very often because it is configured as a "Refinement server" - It only calls sighthound API if your local AI had already detected a vehicle.

**Comment by katoog** [comment]
Excellent thank you, can confirm the error is gone and I'm getting results come through to Telegram with the number plate.

Thanks for the link to sighhound, I'll check it out though I'm always worried about posting images to an online service regardless of how many promises they make around privacy etc. I'm guessing it uploads the whole image rather than just a cropped version of the number plate.

That's good to know about the "Refinement server" functionality. Never understood it's use until now.

Thanks
